contracts/src/v0.1/RequestResponseConsumerBase.sol

```diff
@@ -22,16 +22,85 @@ abstract contract RequestResponseConsumerBase {
      * @return req request in memory
      */
     function buildRequest(bytes32 jobId) internal view returns (Orakl.Request memory req) {
-        return req.initialize(jobId, address(COORDINATOR), COORDINATOR.fulfillDataRequest.selector);
+        bytes4 callbackFunc = detectCallbackFunc(jobId);
+        return req.initialize(jobId, address(COORDINATOR), callbackFunc);
     }
 
-    function fulfillDataRequest(uint256 requestId, uint256 response) internal virtual;
+    function detectCallbackFunc(bytes32 jobId) internal view returns (bytes4 callbackFunc) {
```

Comment:
I think we can make this much better. Currently for every request call we go through this convoluted if-else branch which does a lot of repeated computation. How about having a private mapping which stores the job IDs and its corresponding function selectors. It can be computed once inside of constructor.

```Solidity
mapping(bytes32 => byte4) private sJobIdToFunctionSelector;

constructor(address _requestResponseCoordinator) {
  COORDINATOR = RequestResponseCoordinatorInterface(_requestResponseCoordinator);
  sJobIdToFunctionSelector[keccak256("uint256")] = COORDINATOR.fulfillDataRequestUint256.selector;
  sJobIdToFunctionSelector[keccak256("int256")] = COORDINATOR.fulfillDataRequestInt256.selector;
  sJobIdToFunctionSelector[keccak256("string")] = COORDINATOR.fulfillDataRequestString.selector;
  sJobIdToFunctionSelector[keccak256("bytes32")] = COORDINATOR.fulfillDataRequestBytes32.selector;
  sJobIdToFunctionSelector[keccak256("bytes")] = COORDINATOR.fulfillDataRequestBytes.selector;
}
```

and then

```Solidity
function buildRequest(bytes32 jobId) internal view returns (Orakl.Request memory req) {
  return req.initialize(jobId, address(COORDINATOR), sJobIdToFunctionSelector[jobId]);
}
```